web/src/presentation/components/Modal.js

```diff
@@ -3,6 +3,38 @@ import PropTypes from 'prop-types';
 import ReactDOM from 'react-dom';
 import './styles/Modal.css';
 
+function removeElement(id) {
+  const elem = document.getElementById(id);
+  return elem.parentNode.removeChild(elem);
+}
+
+function createElement(id) {
+  const element = document.createElement('div');
+  element.setAttribute('id', id);
+  return element;
+}
+
+export const showModal = (title, content) => {
+  const id = 'dinamically-generated-modal';
+  const element = createElement(id);
+
+  const unMountAndRemoveElement = () => {
+    ReactDOM.unmountComponentAtNode(element);
+    removeElement(id);
+  };
+
+  ReactDOM.render(
+    <ModalContent
+      title={title}
+      content={content}
+      onClose={unMountAndRemoveElement}
+    />,
+    element,
+  );
+
+  document.body.append(element);
```

Comment:
Para o elemento aparecer na tela, além de renderizar ele precisa estar no documento. Por isso precisar dar `append` no elemento recém criado e recém renderizado.

Reply:
Note que isso é bem parecido com o que acontece em:
src/index.js#10
![image](https://user-images.githubusercontent.com/823150/50546815-71df4f80-0c2e-11e9-9a24-66e335d9a467.png)
 A diferença que estamos usando o `ReactDOM.render` num `node` que já existe no `document`.

public/index.html#15
![image](https://user-images.githubusercontent.com/823150/50546832-91767800-0c2e-11e9-9da9-f1dd84521669.png)

também podiamos fazer isso, criar um novo nó vazio e sempre usar esse nó para renderizar as modais criadas programaticamente.

Reply:
Agora to com vontade de gerar dinamicamente o `div#react-app` 🤓

Reply:
Vai com calma padawan, to entendendo o paranaue ainda xD